src/Core/Validation/ValidationException.php

```diff
@@ -48,6 +52,16 @@ public function __construct($result = null, $code = 0)
             // Pick first message
             foreach ($result->getMessages() as $message) {
                 $exceptionMessage = $message['message'];
+                if ($this->doShowAdditionalInfo()) {
+                    if ($message['fieldName']) {
+                        $exceptionMessage .= ' - fieldName: ' . $message['fieldName'];
+                    }
+                    $dataClass = $result->getDataClass();
+                    if (is_subclass_of($dataClass, DataObject::class, true)) {
+                        $exceptionMessage .= ', recordID: ' . $result->getRecordID();
+                        $exceptionMessage .= ', dataClass: ' . $dataClass;
+                    }
+                }
                 break;
             }
         } elseif (is_string($result)) {
```

Comment:
Here, where only a string is provided and no `ValidationResult`, seems to be where the conditional `$this->doShowAdditionalInfo()` should be used.

We should definitely be adding this extra context here when it's appropriate.

Reply:
I don't understand this. We don't know the extra context if $result is a string, $result is only an error string and there's no way to get the extra info because it's not a ValidationResult object